sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/monitor/applicationinsights/spring/selfdiagnostics/SelfDiagAutoConfig.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.applicationinsights.spring.selfdiagnostics;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+/**
+ Main configuration entry point of the self-diagnostics
+ **/
+@Configuration(proxyBeanMethods = false)
+@Import({DefaultLogConfig.class, LogbackSelfDiagConfig.class, JdbcSelfDiagConfig.class})
+public class SelfDiagAutoConfig {
+    private static final Logger LOG = LoggerFactory.getLogger(SelfDiagAutoConfig.class);
+    private static final String SELF_DIAGNOSTICS_LEVEL_ENV_VAR = "APPLICATIONINSIGHTS_SELF_DIAGNOSTICS_LEVEL";
+
+    @Bean
+    SelfDiagnosticsLevel selfDiagnosticsLevel() {
+        String selfDiagLevelEnvVar = System.getenv(SELF_DIAGNOSTICS_LEVEL_ENV_VAR);
```

Comment:
can we get this from Spring configuration instead?

Reply:
The behavior would be different between the Java agent and this starter. The user could then set the self-diagnostics level with a system property, which is not possible with the Java agent. See https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.

Reply:
> The user could then set the self-diagnostics level with a system property

I think that's ok, and maybe more consistent with expectation of spring boot starter users